src/events.cpp

```diff
@@ -736,6 +753,35 @@ void peek_for_resize()
 	}
 }
 
+void call_in_main_thread(const std::function<void(void)>& f)
+{
+	if (boost::this_thread::get_id() == main_thread) {
+		// nothing special to do if called from the main thread.
+		f();
+		return;
+	}
+
+
+	invoked_funktion_data fdata = { false, f };
+
+	SDL_Event sdl_event;
+	SDL_UserEvent sdl_userevent;
+
+	sdl_userevent.type = INVOKE_FUNCTION_EVENT;
```

Comment:
I'm wondering why this can't be handled with the GUI2/SDL timer system, with a delay of 0. If I understand correctly, that should result in the function being called on the main thread, and it removes the need for a new event type.